lib/api/api-fetch/body.js

```diff
@@ -1,66 +1,178 @@
 'use strict'
 
 const util = require('../../core/util')
-const { Readable } = require('stream')
+const { kState } = require('./symbols')
+const { Blob } = require('buffer')
+const { NotSupportedError } = require('../../core/errors')
+const assert = require('assert')
 
-let TransformStream
+let ReadableStream
 
 // https://fetch.spec.whatwg.org/#concept-bodyinit-extract
 function extractBody (body) {
   // TODO: FormBody
 
+  let stream = null
+  let source = null
+  let length = null
+  let type = null
+
   if (body == null) {
     return [null, null]
   } else if (body instanceof URLSearchParams) {
     // spec says to run application/x-www-form-urlencoded on body.list
     // this is implemented in Node.js as apart of an URLSearchParams instance toString method
     // See: https://github.com/nodejs/node/blob/e46c680bf2b211bbd52cf959ca17ee98c7f657f5/lib/internal/url.js#L490
     // and https://github.com/nodejs/node/blob/e46c680bf2b211bbd52cf959ca17ee98c7f657f5/lib/internal/url.js#L1100
-    return [{
-      source: body.toString()
-    }, 'application/x-www-form-urlencoded;charset=UTF-8']
-  } else if (typeof body === 'string') {
-    return [{
-      source: body
-    }, 'text/plain;charset=UTF-8']
+    source = body.toString()
+    length = Buffer.byteLength(source)
+    type = 'application/x-www-form-urlencoded;charset=UTF-8'
   } else if (body instanceof ArrayBuffer || ArrayBuffer.isView(body)) {
-    return [{
-      source: body
-    }, null]
+    if (body instanceof DataView) {
+      // TODO: Blob doesn't seem to work with DataView?
+      body = body.buffer
+    }
+    source = body
+    length = Buffer.byteLength(body)
   } else if (util.isBlob(body)) {
```

Comment:
We are only supporting fetch in 16+ right?

Reply:
Correct,